cli/tests/activate.bats

```diff
@@ -876,6 +876,24 @@ EOF
 
 # ---------------------------------------------------------------------------- #
 
+# bats test_tags=activate,activate:envVar:zsh:HISTFILE
+@test "zsh: activate inherits HISTFILE" {
+  project_setup
+
+  # TODO: flox will set HOME if it doesn't match the home of the user with
+  # current euid. I'm not sure if we should change that, but for now just set
+  # USER to REAL_USER.
```

Comment:
We do this because Nix otherwise breaks with `sudo`, which will leave the `HOME` variable unchanged unless invoked with the `-H` flag, which then breaks Nix's ability to write to `$HOME` and leads to much unhappiness.

Regarding this test, we need to update the `.zshrc` and `.zshenv` file to do the same thing we find in Apple's `/etc/zshrc`, which is namely to unconditionally set `HISTFILE=$ZDOTDIR/whatever`. We should then test both the login and non-login cases in the interactive context (using expect) and verify that the `HISTFILE` is successfully with some bit of expected output.

Reply:
> we need to update the .zshrc and .zshenv file to do the same thing we find in Apple's /etc/zshrc

Unconditionally setting `HISTFILE=$ZDOTDIR/whatever` was what we started with, do you suggest to add those newly to `assets/activation-scripts/activate.d/zdotdir/.zsh{rc, env}`?
If so, I guess we want that to happen _before_ any user config right, and keep the current logic in `assets/activation-scripts/activate.d/zsh` that may inherit from the user conf?